lib/LaTeXML/Util/Pathname.pm

```diff
@@ -433,6 +433,7 @@ sub build_kpse_cache {
   foreach my $path (split(/$KPATHSEP/, $texpaths)) {
     $path =~ s/^!!//; $path =~ s|//+$|/|;
     push(@filters, $path) if -d $path; }
+  my $filterre = '(?:' . join('|', map { "\Q$_\E"; } @filters) . ')';
```

Comment:
Edge case note: I think there is a subtle behavior change here if `@filters` is empty. The new regex will then evaluate to `(?:)`, which apparently matches against every string. So in that case `$skip` would be always false, rather than always true. I think?